plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/server/behavior/OpenShiftPublishController.java

```diff
@@ -49,8 +50,18 @@
 	@Override
 	public void publishStart(final IProgressMonitor monitor) 
 			throws CoreException {
+		
 		syncDownFailed = false;
-		final IProject deployProject = OpenShiftServerUtils.getDeployProject(getServer());
+		IServer server = getServer();
+		
+		
+		// Noode.js Debug session is running - do *not* perform rsync!
+		if (DebugTrackerContributionEvaluation.isDebugSessionAlive(server)) {
+			return;
```

Comment:
@alexeykazakov agree, let's keep it simple for now and not put the cart before the horse ;-)